### PR TITLE
updated google analytics code

### PR DESCRIPTION
**Pull request by dlebauer**
Updates to new Google Analytics tracking code. Not required, but would be nice to deploy esp to www.betydb.org

could also limit to only checking when on betydb.org

```
if (location.hostname === 'betydb.org' || location.hostname === 'www.betydb.org') {
 <new code>
}
```

### CODE REVIEW

File: app/views/layouts/fullscreen.html.erb [hunk 1]
```diff
@@ -20,23 +20,16 @@
...
-    _gaq.push(['_setAccount', 'UA-30952394-1']);
-    _gaq.push(['_trackPageview']);
-
-    (function() {
-      var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
-      ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
-      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
-    })();
-
+    gtag('config', 'G-2PN4PE3MT2');
```
Comment (robkooper) [inline app/views/layouts/fullscreen.html.erb]: should this be hardcoded in the app, can we make this an option in the config?

File: public/UI_redesign/index.html [hunk 2]
```diff
@@ -456,10 +456,15 @@ <h3><strong>Translate</strong> Page</h3>
 });
 </script>
 
-<!-- http://mths.be/aab -->
+
+<!-- Google Anaytics (GA4) tag (gtag.js) -->
+<script async src="https://www.googletagmanager.com/gtag/js?id=G-2PN4PE3MT2"></script>
```
Comment (robkooper) [inline public/UI_redesign/index.html:461]: can this be a configuration option.

File: public/UI_redesign/index.html [hunk 3]
```diff
@@ -456,10 +456,15 @@ <h3><strong>Translate</strong> Page</h3>
...
 <script>
-var _gaq=[['_setAccount','UA-XXXXX-X'],['_trackPageview']]; // Change UA-XXXXX-X to be your site's ID
-//(function(d,t){var g=d.createElement(t),s=d.getElementsByTagName(t)[0];g.src=('https:'==location.protocol?'//ssl':'//www')+'.google-analytics.com/ga.js';s.parentNode.insertBefore(g,s)}(document,'script'));
+  window.dataLayer = window.dataLayer || [];
+  function gtag(){dataLayer.push(arguments);}
+  gtag('js', new Date());
+
+  gtag('config', 'G-2PN4PE3MT2');
```
Comment (robkooper) [inline public/UI_redesign/index.html:467]: final location can this be a configuration option.

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -93,18 +93,15 @@
...
-    _gaq.push(['_setAccount', 'UA-30952394-1']);
-    _gaq.push(['_trackPageview']);
-
-    (function() {
-      var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
-      ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
-      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
-    })();
+  <!-- Google Analytics (G4)  tag (gtag.js) -->
+  <script async src="https://www.googletagmanager.com/gtag/js?id=G-2PN4PE3MT2"></script>
```
Comment (robkooper) [inline app/views/layouts/application.html.erb]: can this be a configuration option?

**Review by robkooper — CHANGES_REQUESTED** [review]

File: app/views/layouts/fullscreen.html.erb (hunk 1 above)
Comment (dlebauer) [reply app/views/layouts/fullscreen.html.erb]: As it stands this PR restores functionality that was lost when google changed its api. Moving this to a configuration file is a great idea. 

I proposed this in https://github.com/PecanProject/bety/issues/511 but it seems like a heavy lift in the absence of a rails developer (and I closed the issue earlier today after giving up on it / realizing that it is a low priority).

**Comment by robkooper** [comment]
fixed it so it will not have hardcoded google analytics code.

**Comment by dlebauer** [comment]
Thanks - where is / did you update the betydb.org configuration file?

**Pull request merged**